src/app/modules/main/communities/tokens/module.nim

```diff
@@ -80,23 +80,33 @@ proc authenticate(self: Module) =
   else:
     self.controller.authenticateUser()
 
-method airdropCollectibles*(self: Module, communityId: string, collectiblesJsonString: string, walletsJsonString: string) =
-  let collectiblesJson = collectiblesJsonString.parseJson
-  self.tempTokenAndAmountList = @[]
-  for collectible in collectiblesJson:
-    let symbol = collectible["key"].getStr
-    let amount = collectible["amount"].getInt
-    let tokenDto = self.controller.getCommunityTokenBySymbol(communityId, symbol)
-    if tokenDto.tokenType == TokenType.Unknown:
-      error "Can't find token for community", communityId=communityId, symbol=symbol
-      return
-    self.tempTokenAndAmountList.add(CommunityTokenAndAmount(communityToken: tokenDto, amount: amount))
+proc getTokenAndAmountList(self: Module, communityId: string, collectiblesJsonString: string): seq[CommunityTokenAndAmount] =
+  try:
+    let collectiblesJson = collectiblesJsonString.parseJson
+    for collectible in collectiblesJson:
+      let contractUniqueKey = collectible["contractUniqueKey"].getStr
+      let amount = collectible["amount"].getInt
+      let tokenDto = self.controller.findContractByUniqueId(contractUniqueKey)
+      if tokenDto.tokenType == TokenType.Unknown:
+        error "Can't find token for community", contractUniqueKey=contractUniqueKey
+        return @[]
+      result.add(CommunityTokenAndAmount(communityToken: tokenDto, amount: amount))
+  except Exception as e:
+    error "Error getTokenAndAmountList", msg = e.msg
 
+method airdropCollectibles*(self: Module, communityId: string, collectiblesJsonString: string, walletsJsonString: string) =
```

Comment:
Same suggestion here.. will it be difference between collectible and asset info if we have the `contractUniqueKey`?